### PR TITLE
[PROTO-81] Different order of data for each epoch

### PR DESCRIPTION
**Pull request by hamuelna**
Change the seed for each epoch so that all worker have the same order for the same epoch. This is to ensure the sharding logic worker correctly and no worker get data already send to other worker.

Another way is to make it completely random where have each worker have different seed number but I will send this PR for discussion

### CODE REVIEW

File: src/protoplast/scrna/anndata/torch_dataloader.py [hunk 1]
```diff
@@ -186,33 +183,24 @@ def __len__(self):
...
-        return (
-            sum(1 for i in range(len(self.files)) for _ in self.batches[i]) / world_size
-        )
+        return sum(1 for i in range(len(self.files)) for _ in self.batches[i]) / world_size
 
     def __iter__(self):
         self._init_rank()
-
+        logger.debug("Counter value: %d", self.counter)
+        random.seed(self.counter * 100)
```
Comment (tri-dataxight) [inline src/protoplast/scrna/anndata/torch_dataloader.py]: We have a random_seed in the trainer class. Should we mix that "base" seed into this one so the reproducibility is more controllable ?

Comment (hamuelna) [reply src/protoplast/scrna/anndata/torch_dataloader.py]: Lets do that maybe we should do self.counter * self.seed

File: tests/test_anndata_torch_loader.py [hunk 2]
```diff
@@ -284,6 +284,48 @@ def test_load_simple(test_even_h5ad_file: str):
...
+            assert isinstance(data, torch.Tensor)
+            assert not data.is_sparse
+            assert not data.is_sparse_csr
+            total_n_in_epoch += 1
+        assert total_n_in_epoch == loader_len
+
+    assert len(first_batches) == num_epochs
+    are_different_1 = not torch.allclose(first_batches[0], first_batches[1])
+    are_different_2 = not torch.allclose(first_batches[1], first_batches[2])
+    assert are_different_1 and are_different_2, "Data was identical between epochs. Shuffling is not working."
```
Comment (tri-dataxight) [inline tests/test_anndata_torch_loader.py]: How low of a chance where 2 epoches have different batch orders, but cells in individual batches are the same?

**Review by tri-dataxight — APPROVED** [review]
